diff --git a/Common/Core/vtkInformationDataObjectKey.cxx b/Common/Core/vtkInformationDataObjectKey.cxx
index ff88b71..95b315d 100644
--- a/Common/Core/vtkInformationDataObjectKey.cxx
+++ b/Common/Core/vtkInformationDataObjectKey.cxx
@@ -14,7 +14,7 @@
 =========================================================================*/
 #include "vtkInformationDataObjectKey.h"
 
-#include "vtkDataObject.h"
+#include "../DataModel/vtkDataObject.h"
 
 
 //----------------------------------------------------------------------------
