diff --git a/Common/Core/CMakeLists.txt b/Common/Core/CMakeLists.txt
index 8158d53..498734f 100644
--- a/Common/Core/CMakeLists.txt
+++ b/Common/Core/CMakeLists.txt
@@ -123,6 +123,11 @@ SET(Kit_SRCS
   vtkXMLFileOutputWindow.cxx
   )
 
+if(vtkCommonDataModel_ENABLED)
+  include_directories(${VTK_BINARY_DIR}/Common/DataModel)
+  set_property(SOURCE vtkInformationDataObjectKey.cxx PROPERTY COMPILE_DEFINITIONS vtkCommonDataModel_ENABLED)
+endif()
+
 # __int64 seems to be platform specific, turning on if
 if(HAVE_VTK_SIZEOF__INT64)
   list(APPEND Kit_SRCS
diff --git a/Common/Core/vtkInformationDataObjectKey.cxx b/Common/Core/vtkInformationDataObjectKey.cxx
index 95b315d..d8afeea 100644
--- a/Common/Core/vtkInformationDataObjectKey.cxx
+++ b/Common/Core/vtkInformationDataObjectKey.cxx
@@ -14,7 +14,9 @@
 =========================================================================*/
 #include "vtkInformationDataObjectKey.h"
 
-#include "vtkDataObject.h"
+#if defined(vtkCommonDataModel_ENABLED)
+# include "../DataModel/vtkDataObject.h"
+#endif
 
 
 //----------------------------------------------------------------------------
@@ -39,13 +41,19 @@ void vtkInformationDataObjectKey::PrintSelf(ostream& os, vtkIndent indent)
 void vtkInformationDataObjectKey::Set(vtkInformation* info,
                                       vtkDataObject* value)
 {
+#if defined(vtkCommonDataModel_ENABLED)
   this->SetAsObjectBase(info, value);
+#endif
 }
 
 //----------------------------------------------------------------------------
 vtkDataObject* vtkInformationDataObjectKey::Get(vtkInformation* info)
 {
+#if defined(vtkCommonDataModel_ENABLED)
   return static_cast<vtkDataObject *>(this->GetAsObjectBase(info));
+#else
+  return 0;
+#endif
 }
 
 //----------------------------------------------------------------------------
