diff --git a/vtkGenerateVTKConfig.cmake b/vtkGenerateVTKConfig.cmake
index cfe0cfb..8a13763 100644
--- a/vtkGenerateVTKConfig.cmake
+++ b/vtkGenerateVTKConfig.cmake
@@ -9,7 +9,7 @@ SET(_VTK_VERSION_PATCH ${VTK_BUILD_VERSION})
 # We use odd minor numbers for development versions.
 # Use a date for the development patch level.
 IF("${_VTK_VERSION_MINOR}" MATCHES "[13579]$")
-  INCLUDE(${VTK_SOURCE_DIR}/Utilities/kwsys/kwsysDateStamp.cmake)
+  INCLUDE(${VTK_SOURCE_DIR}/Utilities/KWSys/vtksys/kwsysDateStamp.cmake)
   SET(_VTK_VERSION_PATCH
     "${KWSYS_DATE_STAMP_YEAR}${KWSYS_DATE_STAMP_MONTH}${KWSYS_DATE_STAMP_DAY}"
     )
