diff --git a/IO/XML/vtkXMLWriterC.h b/IO/XML/vtkXMLWriterC.h
index 6397c13..9398363 100644
--- a/IO/XML/vtkXMLWriterC.h
+++ b/IO/XML/vtkXMLWriterC.h
@@ -17,17 +17,6 @@
 
 #include "vtkType.h" /* For scalar and vtkDataObject type enumerations.  */
 
-/* Define the export macro.  */
-#if defined(_WIN32) && defined(VTK_BUILD_SHARED_LIBS)
-# if defined(vtkIO_EXPORTS)
-#  define VTK_XML_EXPORT __declspec(dllexport)
-# else
-#  define VTK_XML_EXPORT __declspec(dllimport)
-# endif
-#else
-# define VTK_XML_EXPORT
-#endif
-
 #ifdef __cplusplus
 extern "C"
 {
@@ -276,7 +265,4 @@ void vtkXMLWriterC_Stop(vtkXMLWriterC* self);
 } /* extern "C" */
 #endif /*cplusplus*/
 
-/* Done with export macro.  */
-#undef VTK_XML_EXPORT
-
 #endif
