diff --git a/Parallel/Core/vtkSocketCommunicator.cxx b/Parallel/Core/vtkSocketCommunicator.cxx
index 789306b..349175e 100644
--- a/Parallel/Core/vtkSocketCommunicator.cxx
+++ b/Parallel/Core/vtkSocketCommunicator.cxx
@@ -32,7 +32,7 @@
 // represent the CVS revision number of this file, so the value must
 // be larger than the last revision which used that strategy.
 #define vtkSocketCommunicatorHashId 100 /* MD5 */
-#include "Parallel/vtkSocketCommunicatorHash.h"
+#include "vtkSocketCommunicatorHash.h"
 
 vtkStandardNewMacro(vtkSocketCommunicator);
 vtkCxxSetObjectMacro(vtkSocketCommunicator, Socket, vtkClientSocket);

diff --git a/Parallel/Core/vtkCommunicator.cxx b/Parallel/Core/vtkCommunicator.cxx
index 7014b39..590df79 100644
--- a/Parallel/Core/vtkCommunicator.cxx
+++ b/Parallel/Core/vtkCommunicator.cxx
@@ -25,7 +25,6 @@
 #include "vtkGenericDataObjectReader.h"
 #include "vtkGenericDataObjectWriter.h"
 #include "vtkIdTypeArray.h"
-#include "vtkImageClip.h"
 #include "vtkImageData.h"
 #include "vtkIntArray.h"
 #include "vtkMultiBlockDataSet.h"
@@ -35,8 +34,6 @@
 #include "vtkSmartPointer.h"
 #include "vtkStructuredGrid.h"
 #include "vtkStructuredPoints.h"
-#include "vtkStructuredPointsReader.h"
-#include "vtkStructuredPointsWriter.h"
 #include "vtkTemporalDataSet.h"
 #include "vtkTypeTraits.h"
 #include "vtkUnsignedCharArray.h"
