diff --git a/Rendering/Text/vtkFreeTypeTools.cxx b/Rendering/Text/vtkFreeTypeTools.cxx
index 7de4220..6704aa1 100644
--- a/Rendering/Text/vtkFreeTypeTools.cxx
+++ b/Rendering/Text/vtkFreeTypeTools.cxx
@@ -29,7 +29,7 @@
 #include "FTLibrary.h"
 
 // The embedded fonts
-#include "fonts/vtkEmbeddedFonts.h"
+#include "vtkEmbeddedFonts.h"
 
 #include <stdint.h>
 
diff --git a/Rendering/Text/vtkFreeTypeUtilities.cxx b/Rendering/Text/vtkFreeTypeUtilities.cxx
index 6060136..a519611 100644
--- a/Rendering/Text/vtkFreeTypeUtilities.cxx
+++ b/Rendering/Text/vtkFreeTypeUtilities.cxx
@@ -28,7 +28,7 @@
 
 // The embedded fonts
 
-#include "fonts/vtkEmbeddedFonts.h"
+#include "vtkEmbeddedFonts.h"
 
 #include <sys/stat.h>
 #include <stdint.h>
