diff --git a/Rendering/Context2D/vtkContextActor.h b/Rendering/Context2D/vtkContextActor.h
index f394b2c..0351239 100644
--- a/Rendering/Context2D/vtkContextActor.h
+++ b/Rendering/Context2D/vtkContextActor.h
@@ -44,7 +44,7 @@ public:
 
   // Description:
   // Get the vtkContext2D for the actor.
-  vtkGetNewMacro(Context, vtkContext2D)
+  vtkGetNewMacro(Context, vtkContext2D);
 
   // Description:
   // Get the chart object for the actor.
