diff --git a/Rendering/Text/vtkFreeTypeTools.cxx b/Rendering/Text/vtkFreeTypeTools.cxx
index f6c9610..61e0a03 100644
--- a/Rendering/Text/vtkFreeTypeTools.cxx
+++ b/Rendering/Text/vtkFreeTypeTools.cxx
@@ -29,7 +29,7 @@
 #include "FTLibrary.h"
 
 // The embedded fonts
-#include "fonts/vtkEmbeddedFonts.h"
+#include "vtkEmbeddedFonts.h"
 
 #ifdef FTGL_USE_NAMESPACE
 using namespace ftgl;
diff --git a/Rendering/Text/vtkFreeTypeUtilities.cxx b/Rendering/Text/vtkFreeTypeUtilities.cxx
index 967b90a..3d2d13f 100644
--- a/Rendering/Text/vtkFreeTypeUtilities.cxx
+++ b/Rendering/Text/vtkFreeTypeUtilities.cxx
@@ -28,7 +28,7 @@
 
 // The embedded fonts
 
-#include "fonts/vtkEmbeddedFonts.h"
+#include "vtkEmbeddedFonts.h"
 
 #include <sys/stat.h>
 
