diff --git a/ThirdParty/Cosmo/CMakeLists.txt b/ThirdParty/Cosmo/CMakeLists.txt
index f0d2f86..5bc7f28 100644
--- a/ThirdParty/Cosmo/CMakeLists.txt
+++ b/ThirdParty/Cosmo/CMakeLists.txt
@@ -1,6 +1,10 @@
  
 project(Cosmo)
 
+set(vtkCosmo_THIRD_PARTY 1)
+set(vtkCosmo_LIBRARIES Cosmo)
+vtk_module_impl()
+
 #SET(Cosmo_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
 
 SET(Cosmo_LIBS)
@@ -36,20 +40,7 @@ SET(Cosmo_SOURCES
 INCLUDE_DIRECTORIES (${Cosmo_SOURCE_DIR} ${Cosmo_BINARY_DIR})
 
 VTK_ADD_LIBRARY(Cosmo ${Cosmo_SOURCES})
-TARGET_LINK_LIBRARIES(Cosmo vtksys vtkCommon ${Cosmo_LIBS})
-
-# Apply user-defined properties to the library target. Such as SO Version
-IF(VTK_LIBRARY_PROPERTIES)
-  SET_TARGET_PROPERTIES(Cosmo PROPERTIES ${VTK_LIBRARY_PROPERTIES})
-ENDIF(VTK_LIBRARY_PROPERTIES)
-
-IF(NOT VTK_INSTALL_NO_LIBRARIES)
-  INSTALL(TARGETS Cosmo
-    EXPORT ${VTK_INSTALL_EXPORT_NAME}
-    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
-    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
-    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
-ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
+TARGET_LINK_LIBRARIES(Cosmo ${vtkCommonCore_LIBRARIES} ${Cosmo_LIBS})
 
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   INSTALL(FILES
