diff --git a/ThirdParty/alglib/CMakeLists.txt b/ThirdParty/alglib/CMakeLists.txt
index 15b7ea1..145474e 100644
--- a/ThirdParty/alglib/CMakeLists.txt
+++ b/ThirdParty/alglib/CMakeLists.txt
@@ -1,5 +1,9 @@
 project( alglib )
 
+set(vtkalglib_THIRD_PARTY 1)
+set(vtkalglib_LIBRARIES vtkalglib)
+vtk_module_impl()
+
 cmake_minimum_required( VERSION 2.4 )
 
 if ( NOT LIBRARY_OUTPUT_PATH )
@@ -41,44 +41,17 @@ vtk_add_library( vtkalglib
   ${ALGLIB_SRCS}
 )
 
-# Apply user-defined properties to the library target.
-if ( VTK_LIBRARY_PROPERTIES )
-  set_target_properties( vtkalglib PROPERTIES ${VTK_LIBRARY_PROPERTIES} )
-endif ( VTK_LIBRARY_PROPERTIES )
-
 #
 # Install rules
 #
 
 # Grab install locations from VTK if we are building as part of VTK
-if ( VTK_INSTALL_BIN_DIR_CM24 )
-  set( alglib_install_bin_dir ${VTK_INSTALL_BIN_DIR_CM24} )
-else ( VTK_INSTALL_BIN_DIR_CM24 )
-  set( alglib_install_bin_dir ${CMAKE_INSTALL_PREFIX}/bin )
-endif ( VTK_INSTALL_BIN_DIR_CM24 )
-
-if ( VTK_INSTALL_LIB_DIR_CM24 )
-  set( alglib_install_lib_dir ${VTK_INSTALL_LIB_DIR_CM24} )
-else ( VTK_INSTALL_LIB_DIR_CM24 )
-  set( alglib_install_lib_dir ${CMAKE_INSTALL_PREFIX}/lib )
-endif ( VTK_INSTALL_LIB_DIR_CM24 )
-
 if ( VTK_INSTALL_INCLUDE_DIR_CM24 )
   set( alglib_install_inc_dir ${VTK_INSTALL_INCLUDE_DIR_CM24} )
 else ( VTK_INSTALL_INCLUDE_DIR_CM24 )
   set( alglib_install_inc_dir ${CMAKE_INSTALL_PREFIX}/include )
 endif ( VTK_INSTALL_INCLUDE_DIR_CM24 )
 
-# Library:
-if ( NOT VTK_INSTALL_NO_LIBRARIES )
-  install( TARGETS vtkalglib
-    EXPORT ${VTK_INSTALL_EXPORT_NAME}
-    RUNTIME DESTINATION ${alglib_install_bin_dir} COMPONENT RuntimeLibraries
-    LIBRARY DESTINATION ${alglib_install_lib_dir} COMPONENT RuntimeLibraries
-    ARCHIVE DESTINATION ${alglib_install_lib_dir} COMPONENT Development
-  )
-endif ( NOT VTK_INSTALL_NO_LIBRARIES )
-
 # Header files:
 if ( NOT VTK_INSTALL_NO_DEVELOPMENT )
   install( FILES
