diff --git a/ThirdParty/zlib/vtkzlib/CMakeLists.txt b/ThirdParty/zlib/vtkzlib/CMakeLists.txt
index 6e182be..129fe40 100644
--- a/ThirdParty/zlib/vtkzlib/CMakeLists.txt
+++ b/ThirdParty/zlib/vtkzlib/CMakeLists.txt
@@ -29,19 +29,6 @@ CONFIGURE_FILE(${VTKZLIB_SOURCE_DIR}/zlibDllConfig.h.in
 
 VTK_ADD_LIBRARY(vtkzlib ${ZLIB_SRCS})
 
-# Apply user-defined properties to the library target.
-IF(VTK_LIBRARY_PROPERTIES)
-  SET_TARGET_PROPERTIES(vtkzlib PROPERTIES ${VTK_LIBRARY_PROPERTIES})
-ENDIF(VTK_LIBRARY_PROPERTIES)
-
-IF(NOT VTK_INSTALL_NO_LIBRARIES)
-  INSTALL(TARGETS vtkzlib
-    EXPORT ${VTK_INSTALL_EXPORT_NAME}
-    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
-    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
-    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
-ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
-
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   INSTALL(FILES
     ${VTKZLIB_SOURCE_DIR}/zlib.h
