diff --git a/ThirdParty/ftgl/CMakeLists.txt b/ThirdParty/ftgl/CMakeLists.txt
index 8f8e37d..cad2c38 100644
--- a/ThirdParty/ftgl/CMakeLists.txt
+++ b/ThirdParty/ftgl/CMakeLists.txt
@@ -1,19 +1,17 @@
 PROJECT (VTKFTGL)
 
+set(vtkftgl_THIRD_PARTY 1)
+set(vtkftgl_LIBRARIES vtkftgl)
+set(vtkftgl_INCLUDE_DIRS ${VTKFTGL_SOURCE_DIR}/src)
+vtk_module_impl()
+set(VTK_FREETYPE_LIBRARIES ${vtkfreetype_LIBRARIES})
+
 #
 # Dependency mask
 #
 INCLUDE_REGULAR_EXPRESSION(".*")
 
 #
-# Include path
-#
-INCLUDE_DIRECTORIES (
-  ${VTKFTGL_SOURCE_DIR}/src
-  ${VTKFTGL_BINARY_DIR}
-)
-
-#
 # Collect the required libs
 #
 SET (VTKFTGL_LIBS "")
@@ -149,19 +149,6 @@ SET (VTKFTGL_LIBS "${VTKFTGL_LIBS};${VTK_FREETYPE_LIBRARIES}")
 VTK_ADD_LIBRARY (vtkftgl ${FTGL_SRCS})
 TARGET_LINK_LIBRARIES(vtkftgl ${VTKFTGL_LIBS})
 
-# Apply user-defined properties to the library target.
-IF(VTK_LIBRARY_PROPERTIES)
-  SET_TARGET_PROPERTIES(vtkftgl PROPERTIES ${VTK_LIBRARY_PROPERTIES})
-ENDIF(VTK_LIBRARY_PROPERTIES)
-
-IF(NOT VTK_INSTALL_NO_LIBRARIES)
-  INSTALL(TARGETS vtkftgl
-    EXPORT ${VTK_INSTALL_EXPORT_NAME}
-    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
-    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
-    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
-ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
-
 #
 # Create the configuration file
 #
