diff --git a/ThirdParty/Cosmo/CMakeLists.txt b/ThirdParty/Cosmo/CMakeLists.txt
index f0d2f86..bd76c8c 100644
--- a/ThirdParty/Cosmo/CMakeLists.txt
+++ b/ThirdParty/Cosmo/CMakeLists.txt
@@ -1,6 +1,10 @@
  
 project(Cosmo)
 
+set(vtkCosmo_THIRD_PARTY 1)
+set(vtkCosmo_LIBRARIES Cosmo)
+vtk_module_impl()
+
 #SET(Cosmo_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
 
 SET(Cosmo_LIBS)
@@ -36,7 +39,7 @@ SET(Cosmo_SOURCES
 INCLUDE_DIRECTORIES (${Cosmo_SOURCE_DIR} ${Cosmo_BINARY_DIR})
 
 VTK_ADD_LIBRARY(Cosmo ${Cosmo_SOURCES})
-TARGET_LINK_LIBRARIES(Cosmo vtksys vtkCommon ${Cosmo_LIBS})
+TARGET_LINK_LIBRARIES(Cosmo ${vtkCommonCore_LIBRARIES} ${Cosmo_LIBS})
 
 # Apply user-defined properties to the library target. Such as SO Version
 IF(VTK_LIBRARY_PROPERTIES)
