diff --git a/ThirdParty/ftgl/CMakeLists.txt b/ThirdParty/ftgl/CMakeLists.txt
index 8f8e37d..cad2c38 100644
--- a/ThirdParty/ftgl/CMakeLists.txt
+++ b/ThirdParty/ftgl/CMakeLists.txt
@@ -1,19 +1,16 @@
 PROJECT (VTKFTGL)
 
+set(vtkftgl_LIBRARIES vtkftgl)
+set(vtkftgl_INCLUDE_DIRS ${VTKFTGL_SOURCE_DIR}/src)
+vtk_module_impl()
+set(VTK_FREETYPE_LIBRARIES ${vtkfreetype_LIBRARIES})
+
 #
 # Dependency mask
 #
 INCLUDE_REGULAR_EXPRESSION(".*")
 
 #
-# Include path
-#
-INCLUDE_DIRECTORIES (
-  ${VTKFTGL_SOURCE_DIR}/src
-  ${VTKFTGL_BINARY_DIR}
-)
-
-#
 # Collect the required libs
 #
 SET (VTKFTGL_LIBS "")
