diff --git a/ThirdParty/oggtheora/vtkoggtheora/CMakeLists.txt b/ThirdParty/oggtheora/vtkoggtheora/CMakeLists.txt
index f1faa06..d0f31e3 100644
--- a/ThirdParty/oggtheora/vtkoggtheora/CMakeLists.txt
+++ b/ThirdParty/oggtheora/vtkoggtheora/CMakeLists.txt
@@ -248,20 +248,6 @@ ADD_DEFINITIONS(-DHAVE_CONFIG_H)
 
 VTK_ADD_LIBRARY(vtkoggtheora ${VTKOGGTHEORA_SRCS})
 
-# Apply user-defined properties to the library target.
-IF(VTK_LIBRARY_PROPERTIES)
-  SET_TARGET_PROPERTIES(vtkoggtheora PROPERTIES ${VTK_LIBRARY_PROPERTIES}
-    )
-ENDIF(VTK_LIBRARY_PROPERTIES)
-
-IF(NOT VTK_INSTALL_NO_LIBRARIES)
-  INSTALL(TARGETS vtkoggtheora
-    EXPORT ${VTK_INSTALL_EXPORT_NAME}
-    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
-    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
-    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
-ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
-
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   INSTALL(FILES
     ${VTKOGGTHEORA_SOURCE_DIR}/vtk_oggtheora_mangle.h
