diff --git a/ThirdParty/freetype/vtkfreetype/CMakeLists.txt b/ThirdParty/freetype/vtkfreetype/CMakeLists.txt
index 7573fc3..ceac6fd 100644
--- a/ThirdParty/freetype/vtkfreetype/CMakeLists.txt
+++ b/ThirdParty/freetype/vtkfreetype/CMakeLists.txt
@@ -109,19 +109,6 @@ IF (APPLE AND VTKFREETYPE_MAC_FONT_SUPPORT)
     TARGET_LINK_LIBRARIES (vtkfreetype "-framework ApplicationServices -framework CoreServices")
 ENDIF (APPLE AND VTKFREETYPE_MAC_FONT_SUPPORT)
 
-# Apply user-defined properties to the library target.
-IF(VTK_LIBRARY_PROPERTIES)
-  SET_TARGET_PROPERTIES(vtkfreetype PROPERTIES ${VTK_LIBRARY_PROPERTIES})
-ENDIF(VTK_LIBRARY_PROPERTIES)
-
-IF(NOT VTK_INSTALL_NO_LIBRARIES)
-  INSTALL(TARGETS vtkfreetype
-    EXPORT ${VTK_INSTALL_EXPORT_NAME}
-    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
-    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
-    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
-ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
-
 #
 # Create the configuration file
 #
