diff --git a/ThirdParty/libproj4/vtklibproj4/CMakeLists.txt b/ThirdParty/libproj4/vtklibproj4/CMakeLists.txt
index c6fe9c6..37b3251 100644
--- a/ThirdParty/libproj4/vtklibproj4/CMakeLists.txt
+++ b/ThirdParty/libproj4/vtklibproj4/CMakeLists.txt
@@ -219,7 +219,7 @@ endif ( GSL_FOUND )
 
 configure_file(
   ${libproj4_SOURCE_DIR}/proj_config.h.cmake
-  ${VTK_BINARY_DIR}/proj_config.h
+  ${libproj4_BINARY_DIR}/proj_config.h
 )
 
 # Apply user-defined properties to the library target.
@@ -240,7 +240,7 @@ ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   INSTALL(FILES
     ${libproj4_SOURCE_DIR}/lib_proj.h
-    ${VTK_BINARY_DIR}/proj_config.h
+    ${libproj4_BINARY_DIR}/proj_config.h
     DESTINATION ${VTK_INSTALL_INCLUDE_DIR_CM24}/vtklibproj4
     COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_DEVELOPMENT)
