diff --git a/ThirdParty/exodusII/vtkexodusII/include/exodusII.h b/ThirdParty/exodusII/vtkexodusII/include/exodusII.h
index 7223b6a..8d42954 100644
--- a/ThirdParty/exodusII/vtkexodusII/include/exodusII.h
+++ b/ThirdParty/exodusII/vtkexodusII/include/exodusII.h
@@ -45,7 +45,7 @@
 
 #include "vtk_netcdf.h"
 #include "vtk_exodus2_mangle.h"
-#include "vtkexodus2/include/exodusII_cfg.h"
+#include "vtkexodusII/include/exodusII_cfg.h"
 
 /* EXODUS II version number */
 #define EX_API_VERS 5.10f
