diff --git a/Utilities/vtknetcdf/CMakeLists.txt b/Utilities/vtknetcdf/CMakeLists.txt
index bbcedfe..4d15929 100644
--- a/Utilities/vtknetcdf/CMakeLists.txt
+++ b/Utilities/vtknetcdf/CMakeLists.txt
@@ -301,6 +301,7 @@ INCLUDE_DIRECTORIES(
   ${CMAKE_CURRENT_SOURCE_DIR}/libsrc4
   ${CMAKE_CURRENT_BINARY_DIR}
   ${VTK_HDF5_INCLUDE_DIR}
+  ${VTK_SOURCE_DIR}/Utilities
   )
 
 set(netcdf_liblib_SOURCES "" CACHE INTERNAL "" FORCE)
